_posts/2020-01-20-graphql-relay-windowed-pagination.markdown

```diff
@@ -0,0 +1,304 @@
+---
+layout: epic
+title: Effortless Pagination with GraphQL and Relay? Really!
+date: 2020-01-20
+categories: [GraphQL, Relay, Pagination]
+author: matt
+---
+
+It's the year 2020. You use a modern front-end stack of [Relay](https://relay.dev/),
+[GraphQL](https://graphql.org/), [React](https://reactjs.org/) and [Typescript](https://www.typescriptlang.org/).
+You can build an infinite scroll 'feed' type UI totally out of the box with these tools, by mostly putting together
+boilerplate (proper [connections](https://facebook.github.io/relay/graphql/connections.htm#sec-Connection-Types),
+along with a [pagination container](https://relay.dev/docs/en/pagination-container)). You have a design system, and
+are rapidly building up a component library. Things are great!
+
+Then you take a look at the latest design comps for a 'browse' type page, and you see that the
+[controversial](https://medium.com/simple-human/7-reasons-why-infinite-scrolling-is-probably-a-bad-idea-a0139e13c96b)
+infinite scroll has been replaced by a more traditional pagination bar.
+
+You know the one. Like the following, from [Amazon](https://www.amazon.com):
+
+<img src="/images/2020-01-20-graphql-relay-windowed-pagination/amazon.png">
+
+You start to realize that the cursor-based setup of a connection, along with a Relay pagination container, does not
+lend itself to this more traditional UI. For one thing, a user can arbitrarily 'jump' to any page by including a
+`?page=X` query param (typically). For another, the user can only actually see the current page of content, vs a
+feed. As you start to dream of REST, Rails controllers, [kaminari](https://github.com/kaminari/kaminari),
+[will_paginate](https://github.com/mislav/will_paginate), and a simpler time, you start to have a vision...
+
+<!-- more -->
+
+To get a good primer of what a GraphQL connection is and why they're so useful, read this
+[excellent Apollo blogpost](https://blog.apollographql.com/explaining-graphql-connections-c48b7c3d6976). Seriously.
+It's one of the best writeups on this subject out there. I'll assume basic familiarity with connection types from
+this point forward.
+
+**To summarize:** We prefer to use connections in place of lists almost always. Not only do they provide a
```

Comment:
"To summarize" placed here seems like the entry is suddenly ending. Maybe just leave the link and caveat and start with "We prefer..."?